install.sh

```diff
@@ -1,41 +1,57 @@
 #!/usr/bin/env bash
+
 set -e
 
-resolve_link() {
-  $(type -p greadlink readlink | head -n1) "$1"
+PREFIX="$1"
+if [ -z "$1" ]; then
+  { echo "usage: $0 <prefix>"
+    echo "  e.g. $0 /usr/local"
+  } >&2
+  exit 1
+fi
+
+export BATS_READLINK='true'
+if command -v 'greadlink' >/dev/null; then
+  BATS_READLINK='greadlink'
+elif command -v 'readlink' >/dev/null; then
+  BATS_READLINK='readlink'
+fi
+
+bats_resolve_link() {
+  if ! "$BATS_READLINK" "$1"; then
+    return 0
+  fi
 }
 
-abs_dirname() {
-  local cwd="$(pwd)"
+bats_absolute_path() {
+  local cwd="$PWD"
   local path="$1"
 
-  while [ -n "$path" ]; do
+  while [[ -n "$path" ]]; do
     cd "${path%/*}"
     local name="${path##*/}"
-    path="$(resolve_link "$name" || true)"
+    path="$(bats_resolve_link "$name")"
   done
 
-  pwd
+  printf -v "$2" -- '%s' "$PWD"
   cd "$cwd"
 }
 
-PREFIX="$1"
-if [ -z "$1" ]; then
-  { echo "usage: $0 <prefix>"
-    echo "  e.g. $0 /usr/local"
-  } >&2
-  exit 1
-fi
+bats_install_scripts() {
+  local scripts_dir
```

Comment:
Style question, any reason not to one-line the locals?

Reply:
In my old age, I've begun to favor "letting the code breathe" by splitting things across multiple lines. Which is to say, I used to optimize for vertical space in several languages, only to curse myself later for being so clever that I had to spend a disproportionate amount of time and mental energy unpacking my compact expressions when I had to revisit them later.

Granted, putting a couple of locals on the same line isn't a big deal, but I just split them to be consistent.

Reply:
Yep. My admittedly narrow opinion is to distinguish "noise" from signal in the code. Variable declarations qualify as noise to me since they aren't really interesting, so I try to compress the amount of space that noise takes up, which in turn let's me more easily uncompress the meaningful parts of the code and let the _signal_ breathe 😄